### PR TITLE
improve(ClientHelper): Cache SpokePool registration blocks

### PR DESCRIPTION
**Pull request by pxrl**
The registration block for each SpokePool deployment is searched on each bot run. These values don't change, so cache them after 1 week.

### CODE REVIEW

**Review by nicholaspai — APPROVED** [review]

File: src/common/ClientHelper.ts [hunk 1]
```diff
@@ -40,6 +40,47 @@ async function getSpokePoolSigners(
...
+  const blockFinder = undefined;
+  const mainnetActivationBlock = hubPoolClient.getSpokePoolActivationBlock(chainId, spokePoolAddr);
+  const { timestamp } = await hubPoolClient.hubPool.provider.getBlock(mainnetActivationBlock);
+  const hints = { lowBlock: getDeploymentBlockNumber("SpokePool", chainId) };
+  const activationBlock = await getBlockForTimestamp(chainId, timestamp, blockFinder, redis, hints);
+
+  const day = 24 * 3600;
+  const minAge = 7 * day;
+  if (isDefined(redis) && getCurrentTime() - timestamp > minAge) {
+    const ttl = Math.round(30 * day - Math.random() * day); // ttl 29 - 30 days.
```
Comment (james-a-morris) [inline src/common/ClientHelper.ts:77]: Should we always have this set to 29 or 30 days?

Comment (mrice32) [reply src/common/ClientHelper.ts:78]: Sorry, why specify this TTL at all? Why not just let it be stored indefinitely until/if redis prunes for space (using its LRU algo)?

File: src/relayer/RelayerClientHelper.ts [hunk 2]
```diff
@@ -49,25 +42,14 @@ async function indexedSpokePoolClient(
 
   // Set up Spoke signers and connect them to spoke pool contract objects.
   const signer = baseSigner.connect(await getProvider(chainId));
-
-  const blockFinder = undefined;
```
Comment (mrice32) [inline src/relayer/RelayerClientHelper.ts:53]: +1 for factoring this out and simplifying this

**Review by mrice32 — COMMENTED** [review]
Only question is around the TTL for the cache

File: src/common/ClientHelper.ts (hunk 1 above)
Comment (pxrl) [reply src/common/ClientHelper.ts:78]: Yeah, can do. I have an instinctive drive to force expiry but there's no real requirement for it here...it's a bad habit that I should shake.

Comment (pxrl) [reply src/common/ClientHelper.ts]: Implemented: https://github.com/across-protocol/relayer-v3/pull/1450/commits/32412487495c507c0b992ec65a6f7280cb4ab943

File: src/common/ClientHelper.ts [hunk 3]
```diff
@@ -40,6 +40,46 @@ async function getSpokePoolSigners(
...
+): Promise<number> {
+  const spokePoolAddr = hubPoolClient.getSpokePoolForBlock(chainId, blockNumber);
+  const key = `relayer_${chainId}_spokepool_${spokePoolAddr}_activation_block`;
+
+  const redis = await getRedisCache(hubPoolClient.logger);
+  if (isDefined(redis)) {
+    const activationBlock = await redis.get(key);
+    const numericActivationBlock = Number(activationBlock);
+    if (Number.isInteger(numericActivationBlock) && numericActivationBlock > 0) {
+      return Number(activationBlock);
```
Comment (pxrl) [inline src/common/ClientHelper.ts:63]:
```suggestion
      return numericActivationBlock;
```

**Pull request merged**